plugins/ipam/commands/commands.go

```diff
@@ -100,17 +100,28 @@ func Del(args *skel.CmdArgs) error {
 }
 
 func del(ipManager ipstore.IPAllocator, ipamConf *config.IPAMConfig) error {
-	if ipamConf.IPV4Address.IP == nil {
-		return errors.New("del commands: ip address is required for deletion")
-	}
+	var releasedIP string
 
-	err := ipManager.Release(ipamConf.IPV4Address.IP.String())
-	if err != nil {
-		return err
+	if ipamConf.IPV4Address.IP != nil {
+		// Release the ip by ip address
+		err := ipManager.Release(ipamConf.IPV4Address.IP.String())
+		if err != nil {
+			return err
+		}
+		releasedIP = ipamConf.IPV4Address.IP.String()
+	} else if ipamConf.ID != "" {
```

Comment:
I think `ipamConf.ID != nil && ipamConf.ID == ""` is a stronger check since this is [not a pointer type](https://github.com/richardpen/amazon-ecs-cni-plugins-1/blob/eae4a48d251e6481aedaa7ec9f6564efdbad5f33/plugins/ipam/config/config.go#L47).